mllib/src/main/scala/org/apache/spark/ml/source/libsvm/LibSVMOptions.scala

```diff
@@ -41,11 +41,15 @@ private[libsvm] class LibSVMOptions(@transient private val parameters: CaseInsen
     case o => throw new IllegalArgumentException(s"Invalid value `$o` for parameter " +
       s"`$VECTOR_TYPE`. Expected types are `sparse` and `dense`.")
   }
+
+  val lineSeparator: Option[String] = parameters.get(LINE_SEPARATOR)
```

Comment:
Yea, actually in case of Univocity, it requires 2 characters:

```
CsvWriterSettings settings = new CsvWriterSettings();
settings.getFormat().setLineSeparator("aaa");
```

```
Exception in thread "main" java.lang.IllegalArgumentException: Invalid line separator. Up to 2 characters are expected. Got 3 characters.
	at com.univocity.parsers.common.Format.setLineSeparator(Format.java:121)
	at com.univocity.parsers.common.Format.setLineSeparator(Format.java:109)
```

I don't see a reason to restrict this for now. At least, I can provide an usecase with Windows -`\r\n`.